src/components/PermalinkField/PermalinkField.js

```diff
@@ -0,0 +1,146 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { MdContentCopy } from 'react-icons/md';
+import Button from '../Button';
+
+const propTypes = {
+  /**
+   * Value of the permalink
+   */
+  value: PropTypes.string,
+
+  /**
+   * Set input field as read only.
+   */
+  readOnly: PropTypes.bool,
+
+  /**
+   * CSS class of the container (input+button).
+   */
+  className: PropTypes.string,
+
+  /**
+   * CSS class of the input field.
+   */
+  classNameInputField: PropTypes.string,
+
+  /**
+   * CSS class of the copy button.
+   */
+  classNameCopyBt: PropTypes.string,
+
+  /**
+   * Content for copy button.
+   */
+  button: PropTypes.any,
+
+  /**
+   * Title for the copy button.
+   */
+  titleCopyBt: PropTypes.string,
+
+  /**
+   * Title for the input field.
+   */
+  titleInputField: PropTypes.string,
+
+  /**
+   * Callback function to get a shortened URL.
+   */
+  getShortenedUrl: PropTypes.func,
+};
+
+const defaultProps = {
+  value: '',
+  readOnly: false,
+  className: 'tm-permalink-field',
+  classNameInputField: 'tm-permalink-input',
+  classNameCopyBt: 'tm-permalink-bt',
+  titleCopyBt: '',
+  titleInputField: '',
+  button: <MdContentCopy focusable={false} />,
+  getShortenedUrl: null,
+};
+
+/**
+ * This component displays a permalink field.
+ */
+class PermalinkField extends PureComponent {
+  constructor(props) {
+    super(props);
+    const { value, getShortenedUrl } = this.props;
+    this.state = {
+      permalinkValue: getShortenedUrl ? getShortenedUrl(value) : value,
+    };
+    this.inputRef = null;
+  }
+
+  componentDidUpdate(prevProps) {
+    this.updatePermalinkValue(prevProps);
+  }
+
+  onClickCopyButton() {
+    if (this.inputRef) {
+      this.inputRef.select();
+    }
+    document.execCommand('copy');
+  }
+
+  updatePermalinkValue(prevProps) {
+    const { value, getShortenedUrl } = this.props;
+
+    if (value !== prevProps.value) {
+      if (getShortenedUrl) {
+        this.setState({ permalinkValue: getShortenedUrl(value) });
```

Comment:
If getShortenedUrl function calls a backend api the result will not be immediat so  setStae must probably be called after the resolve of a promise.